packages/neon_framework/packages/neon_http_client/lib/src/interceptors/authorization_throttling_interceptor.dart

```diff
@@ -49,15 +52,52 @@ final class AuthorizationThrottlingInterceptor implements HttpInterceptor {
   }
 
   @override
-  StreamedResponse interceptResponse({required StreamedResponse response, required Uri url}) {
+  Future<StreamedResponse> interceptResponse({required StreamedResponse response, required Uri url}) async {
     assert(
       shouldInterceptResponse(response),
       'Response should not be intercepted.',
     );
 
     final authorization = response.request!.headers['authorization'];
     if (authorization != null && authorization.isNotEmpty && response.statusCode == 401) {
-      _blocked = true;
+      final contentType = response.headers['content-type'];
+      if (contentType == null) {
+        return response;
+      }
+
+      final mediaType = MediaType.parse(contentType);
+      if (mediaType.mimeType != 'application/json') {
+        return response;
+      }
+
+      final syncResponse = await Response.fromStream(response);
+      try {
+        final jsonBody = json.decode(syncResponse.body);
+        if (jsonBody is Map<String, dynamic>) {
+          final ocs = jsonBody['ocs'];
+          if (ocs is Map<String, dynamic>) {
+            final ocsMeta = ocs['meta'];
+            if (ocsMeta is Map<String, dynamic>) {
+              final ocsMetaStatusCode = ocsMeta['statuscode'];
+              if (ocsMetaStatusCode == 997) {
+                _blocked = true;
+              }
+            }
+          }
+        }
```

Comment:
Pattern matching should help (https://dart.dev/language/pattern-types#map):
```suggestion
        if (jsonBody case {'ocs': {'meta': {'statuscode': 997}}}) {
          _blocked = true;
        }
```

I hope there isn't any typo, but something like the above might do the trick.

Reply:
Works :heart_eyes: